src/Servers/HttpSys/test/FunctionalTests/HttpsTests.cs

```diff
@@ -160,6 +163,46 @@ public async Task Https_SetsITlsHandshakeFeature()
             }
         }
 
+        [ConditionalFact]
+        [OSSkipCondition(OperatingSystems.Windows, WindowsVersions.Win7, WindowsVersions.Win2008R2)]
+        public async Task Https_ITlsHandshakeFeature_MatchesIHttpSysExtensionInfoFeature()
+        {
+            using (Utilities.CreateDynamicHttpsServer(out var address, async httpContext =>
+            {
+                try
+                {
+                    var tlsFeature = httpContext.Features.Get<ITlsHandshakeFeature>();
+                    var requestInfoFeature = httpContext.Features.Get<IHttpSysRequestInfoFeature>();
+                    Assert.NotNull(tlsFeature);
+                    Assert.NotNull(requestInfoFeature);
+                    Assert.True(requestInfoFeature.RequestInfo.Count > 0);
+                    var tlsInfo = requestInfoFeature.RequestInfo.Where(info => info.Key == (int)HttpApiTypes.HTTP_REQUEST_INFO_TYPE.HttpRequestInfoTypeSslProtocol).Single().Value;
+                    HttpApiTypes.HTTP_SSL_PROTOCOL_INFO tlsCopy;
+                    unsafe
+                    {
+                        using var handle = tlsInfo.Pin();
+                        tlsCopy = Marshal.PtrToStructure<HttpApiTypes.HTTP_SSL_PROTOCOL_INFO>((IntPtr)handle.Pointer);
+                    }
+
+                    // Assert.Equal(tlsFeature.Protocol, tlsCopy.Protocol); // These don't directly match because there's some value mapping that needs to happen.
```

Comment:
Clarify on this?

Reply:
The managed and native enum values don't directly align.
https://github.com/aspnet/AspNetCore/blob/921816263779c158e3d1d76591d63d928cc9c669/src/Servers/HttpSys/src/RequestProcessing/Request.cs#L260-L285